pkg/osquery/tables/execparsers/remotectl/parse.go

```diff
@@ -0,0 +1,259 @@
+
+	for p.scanner.Scan() {
+		p.lastReadLine = strings.TrimSpace(p.scanner.Text())
+
+		// Exiting dictionary
+		if p.lastReadLine == "}" {
+			return dictionaryResults, nil
+		}
+
+		propertyKey, propertyValue, err := extractPropertyKeyValue(p.lastReadLine)
+		if err != nil {
+			return nil, err
+		}
+
+		dictionaryResults[propertyKey] = propertyValue
+	}
+
+	return nil, errors.New("unexpected end to dictionary")
+}
+
+func (p *parser) parseStringArray() ([]string, error) {
+	arrayResults := make([]string, 0)
+
+	startingIndentationLevel := p.getCurrentIndentationLevel()
+	for p.scanner.Scan() {
+		p.lastReadLine = p.scanner.Text()
+
+		currentIndentationLevel := p.getCurrentIndentationLevel()
+
+		// Exiting array
+		if currentIndentationLevel <= startingIndentationLevel {
+			return arrayResults, nil
+		}
+
+		// Ignore everything not at the top level
+		if currentIndentationLevel == startingIndentationLevel+1 {
+			arrayResults = append(arrayResults, strings.TrimSpace(p.lastReadLine))
+		}
+	}
+
+	return arrayResults, nil
+}
+
+func (p *parser) parseObjectArray() ([]map[string]interface{}, bool, error) {
+	arrayResults := make([]map[string]interface{}, 0)
+	eof := false
+
+	startingIndentationLevel := p.getCurrentIndentationLevel()
+	arrayItemIndentationLevel := startingIndentationLevel + 1
+	arrayItemPropertyIndentationLevel := arrayItemIndentationLevel + 1
+
+	for p.scanner.Scan() {
+		p.lastReadLine = p.scanner.Text()
+
+		currentIndentationLevel := p.getCurrentIndentationLevel()
+
+		// Exiting array
+		if currentIndentationLevel <= startingIndentationLevel {
+			return arrayResults, eof, nil
+		}
+
+		// Process items
+		if currentIndentationLevel == arrayItemIndentationLevel {
+			item := make(map[string]interface{})
+			// Create artificial key "Name" to hold the name of the item
+			item["Name"] = strings.TrimSpace(p.lastReadLine)
+			arrayResults = append(arrayResults, item)
+			continue
+		}
+
+		// One more level indented -- we have properties attached to the item we processed last. Extract them.
+		if currentIndentationLevel >= arrayItemPropertyIndentationLevel {
+			lastProcessedItem := arrayResults[len(arrayResults)-1]
+
+			if strings.HasPrefix(strings.TrimSpace(p.lastReadLine), "Properties:") {
+				itemProperties, err := p.parseDictionary()
+				if err != nil {
+					return nil, eof, err
+				}
+				lastProcessedItem["Properties"] = itemProperties
+				continue
+			}
+
+			propertyKey, propertyValue, err := extractTopLevelKeyValue(p.lastReadLine)
+			if err != nil {
+				return nil, eof, err
+			}
+			lastProcessedItem[propertyKey] = propertyValue
+		}
+	}
+
+	eof = true
+	return arrayResults, eof, nil
+}
+
+func (p *parser) getCurrentIndentationLevel() int {
+	return strings.LastIndex(p.lastReadLine, "\t")
+}
+
+func extractPropertyKeyValue(line string) (string, string, error) {
+	// key-value pairs in the `Properties` dictionary are in the format `key => value`
+	extracted := strings.Split(line, "=>")
```

Comment:
nit: Since these functions are the same except for the delimiter they split strings on, they could both be wrappers for one function which parameterizes the delimiter.

Reply:
Nice suggestion, thank you -- updated!